components/SalesHistoryTable/index.tsx

```diff
@@ -76,14 +123,20 @@ const SalesHistoryTable = ({ tableData, error }: Props): JSX.Element => {
 
   useEffect(() => {
     (async () => {
-      if (tableData.length) {
-        const chainAccounts = tableData.map(({ buyer }) => buyer);
-        const res = await getAvatars(chainAccounts);
-        setAvatars(res);
+      try {
+        if (renderedData.length) {
+          const chainAccounts = renderedData.map(({ buyer }) => buyer);
+          const res = await getAvatars(chainAccounts);
+          setAvatars(res);
+        }
+        router.prefetch('/');
+        await prefetchNextPage();
+        setIsLoading(false);
+      } catch (e) {
+        setErrorMessage(e.message);
```

Comment:
I believe we'll also need to setIsLoading(false) even if it goes into the error so that it won't be loading forever.